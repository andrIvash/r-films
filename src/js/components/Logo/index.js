// @flow
import React from 'react';
import { Link } from 'react-router-dom';

const Logo = () => (
  <div className='logo app__logo'>
    <Link to={{ pathname: '/' }}>
      netflixroulette
    </Link>
  </div>
);

export default Logo;
