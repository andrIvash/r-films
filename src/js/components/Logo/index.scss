.logo {
  a {
    font-size: rem(16px);
    font-family: $sign-font;
    font-weight: 700;
    text-decoration: none;
    color: $color-logo; 
    &:hover, &:active {
      color: $color-logo-dark;
    }
  }
}