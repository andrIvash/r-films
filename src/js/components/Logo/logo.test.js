import React from 'react';
import Logo from './index';

jest.mock('react-router-dom/NavLink', () => 'nav-link');
jest.mock('react-router-dom/Link', () => 'router-link');

describe('Logo', () => {
  it('should be defined', () => {
    expect(Logo).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = render(
      <Logo /> );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a proper logo', () => {
    const wrapper = shallow(
      <Logo /> );
    expect(wrapper.find('.logo').text())
      .toBe('netflixroulette');
  });
});
