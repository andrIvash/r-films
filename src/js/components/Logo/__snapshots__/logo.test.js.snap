// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render correctly 1`] = `
<div
  class="logo app__logo"
>
  netflixroulette
</div>
`;
